src/cljs/main/broadfcui/page/workspace/data/import_data.cljs

```diff
@@ -15,21 +15,27 @@
 (def ^:private preview-limit 4096)
 
 (react/defc Page
-  {:render
+  {:get-initial-state
+   (fn []
+     {:file-input-key (gensym "file-input-")})
+   :render
    (fn [{:keys [state refs this]}]
      [:div {:style {:textAlign "center"}}
       (when (:loading? @state)
         [comps/Blocker {:banner "Uploading file..."}])
-
-      [:input {:type "file" :name "entities" :ref "entities"
+      [:input {:key (:file-input-key @state)
```

Comment:
It isn't obvious why we need to apply this technique to this input element. I suggest a comment, such as:
```cljs
;; This key is changed every time a file is selected causing React to completely replace the
;; element. Otherwise, if a user selects the same file (even after having modified it), the
;; browser will not fire the onChange event.
```